src/features/hotspots/settings/HotspotSettingsProvider.tsx

```diff
@@ -0,0 +1,48 @@
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useRef,
+  useState,
+} from 'react'
+
+const initialState = {
+  showBack: false,
+  enableBack: () => {},
+  disableBack: () => {},
+  goBack: () => {},
+}
+
+const useHotspotSettings = () => {
+  const [showBack, setShowBack] = useState(false)
+  const backHandler = useRef<() => void | undefined>()
+
+  const enableBack = (handler: () => void) => {
+    setShowBack(true)
+    backHandler.current = handler
+  }
+
+  const disableBack = () => {
+    setShowBack(false)
+  }
+
+  const goBack = () => {
+    backHandler?.current?.()
+  }
+
+  return { showBack, enableBack, disableBack, goBack }
+}
+
+const HotspotSettingsContext = createContext<
```

Comment:
one thing I was curious is how we're differentiating between what should be a context/provider vs a redux slice? In my branch, I started breaking some things out into more granular slices. One for activity etc. I could see this being a hotspot settings slice. is there any reason to pick provider/context over redux? or should we start trying to centralize on redux?

Reply:
I think you're right that we should centralize on redux by default. For this instance, I used a context because storing functions in redux is frowned upon. I'm sure it would work though. I probably could have architected this differently and used redux to track the overall navigation state for HotspotSettings. I took the approach of allowing each "sub" view to manage their own navigation.

Reply:
ok that's fine, we'll just need to see how this plays with the changes i made in bottom sheet